src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -52,6 +55,165 @@ @interface GRPCCall ()<GRXWriteable>
+    [NSException raise:NSInvalidArgumentException format:@"Neither host nor path can be nil."];
+  }
+
+  if ((self = [super init])) {
+    _requestOptions = [requestOptions copy];
+    _callOptions = [callOptions copy];
+    _handler = handler;
+    _initialMetadataPublished = NO;
+    _pipe = [GRXBufferedPipe pipe];
+    _dispatchQueue = dispatch_queue_create(NULL, DISPATCH_QUEUE_SERIAL);
+  }
+
+  return self;
+}
+
+- (instancetype)initWithRequestOptions:(GRPCRequestOptions *)requestOptions
+                               handler:(id<GRPCResponseHandler>)handler {
+  return [self initWithRequestOptions:requestOptions handler:handler callOptions:nil];
+}
+
+- (void)start {
+  dispatch_async(_dispatchQueue, ^{
+    if (!self->_callOptions) {
+      self->_callOptions = [[GRPCCallOptions alloc] init];
+    }
+
+    self->_call = [[GRPCCall alloc] initWithHost:self->_requestOptions.host
+                                            path:self->_requestOptions.path
+                                      callSafety:self->_requestOptions.safety
+                                  requestsWriter:self->_pipe
+                                     callOptions:self->_callOptions];
+    if (self->_callOptions.initialMetadata) {
+      [self->_call.requestHeaders addEntriesFromDictionary:self->_callOptions.initialMetadata];
+    }
+    id<GRXWriteable> responseWriteable = [[GRXWriteable alloc] initWithValueHandler:^(id value) {
+      dispatch_async(self->_dispatchQueue, ^{
+        if (self->_handler) {
+          id<GRPCResponseHandler> handler = self->_handler;
+          NSDictionary *headers = nil;
+          if (!self->_initialMetadataPublished) {
+            headers = self->_call.responseHeaders;
+            self->_initialMetadataPublished = YES;
+          }
+          if (headers) {
+            dispatch_async(handler.dispatchQueue, ^{
+              [handler receivedInitialMetadata:headers];
+            });
+          }
+          if (value) {
+            dispatch_async(handler.dispatchQueue, ^{
+              [handler receivedMessage:value];
+            });
+          }
+        }
+      });
+    }
+        completionHandler:^(NSError *errorOrNil) {
+          dispatch_async(self->_dispatchQueue, ^{
+            if (self->_handler) {
+              id<GRPCResponseHandler> handler = self->_handler;
+              NSDictionary *headers = nil;
+              if (!self->_initialMetadataPublished) {
+                headers = self->_call.responseHeaders;
+                self->_initialMetadataPublished = YES;
+              }
+              if (headers) {
+                dispatch_async(handler.dispatchQueue, ^{
+                  [handler receivedInitialMetadata:headers];
+                });
+              }
+              dispatch_async(handler.dispatchQueue, ^{
+                [handler closedWithTrailingMetadata:self->_call.responseTrailers error:errorOrNil];
+              });
+            }
```

Comment:
Does this need to clear self->_handler and self->_call to properly interact with the logic in cancel?